pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go

```diff
@@ -56,59 +50,42 @@ type NamespaceController struct {
 	getData func() map[string]string
 	client  corev1.CoreV1Interface
 
-	queue queue.Instance
-
-	// Controller and store for namespace objects
-	namespaceController cache.Controller
-	// Controller and store for ConfigMap objects
-	configMapController cache.Controller
+	queue              queue.Instance
+	namespacesInformer cache.SharedInformer
+	configMapInformer  cache.SharedInformer
 }
 
 // NewNamespaceController returns a pointer to a newly constructed NamespaceController instance.
-func NewNamespaceController(data func() map[string]string, options Options, kubeClient kubernetes.Interface) *NamespaceController {
+func NewNamespaceController(data func() map[string]string, kubeClient kube.Client) *NamespaceController {
 	c := &NamespaceController{
 		getData: data,
 		client:  kubeClient.CoreV1(),
 		queue:   queue.NewQueue(time.Second),
 	}
 
-	watchedNamespaceList := strings.Split(options.WatchedNamespaces, ",")
-
-	mlw := listwatch.MultiNamespaceListerWatcher(watchedNamespaceList, func(namespace string) cache.ListerWatcher {
-		return &cache.ListWatch{
-			ListFunc: func(opts metav1.ListOptions) (runtime.Object, error) {
-				opts.LabelSelector = fields.SelectorFromSet(configMapLabel).String()
-				return kubeClient.CoreV1().ConfigMaps(namespace).List(context.TODO(), opts)
```

Comment:
just realized this watch only the specific configmap, with this pr we watch all. The difference lays on the memory store.